internal/lobby/routes.go

```diff
@@ -29,6 +29,7 @@ func NewLobbiesRouter() http.Handler {
 	mux.HandleFunc("GET /lobbies/{$}", getLobbiesHandler)
 	mux.HandleFunc("GET /lobbies/{pin}", getLobbyByPinHandler)
 	mux.HandleFunc("/lobbies/{pin}/ws", getLobbyByPinWsHandler)
+	mux.HandleFunc("/lobbies/leaderboard", getLeaderboardHandler)
```

Comment:
Move this handler to package "results" (you need to create it) and change the path to "/results/{pin}"